project3.cc

```diff
@@ -0,0 +1,599 @@
+//TODO add stuff here
+
+#include "project3.h"
+#include <iomanip>
+using namespace std;
+
+/*
+ * Methods of class ThreadedBinarySearchTree
+ *
+ */
+
+typedef ThreadedBinarySearchTree::Iterator   Iterator;
+
+ThreadedBinarySearchTree::ThreadedBinarySearchTree()
+  : _header(new  Node())
+{
+  _header->_lchild = makeThread(_header);
+  _header->_rchild = _header;
+}
+
+/*
+* Destructor called when the user clicks q in the program
+* Iterators through the program deleting all nodes
+* To ensure we are a good steward of nodes
+*/
+ThreadedBinarySearchTree::~ThreadedBinarySearchTree()
+{
+  //Adam and Dr Tuck
+  Iterator it;
+  Node *tempnode;
+
+  for (it = this->inorder_begin(); it != this->end(); it++)
+  {
+    //temp node = to pointer
+    tempnode = it._ptr;
+    delete it._ptr;
```

Comment:
This deletes it._ptr before it is used by the iterator (via "it++" in for loop).  
"Use after delete" is a classic and pernicious bug.  
It's easy to do, and can cause strange symptoms that are hard to debug.